src/v/storage/segment_reader.h

```diff
@@ -62,8 +62,9 @@ class segment_reader {
 
     /// create an input stream _sharing_ the underlying file handle
     /// starting at position @pos
+    /// limit the stream size by setting @read_size (0 = whole file)
     ss::input_stream<char>
-    data_stream(size_t pos, const ss::io_priority_class&);
+    data_stream(size_t pos, const ss::io_priority_class&, size_t read_size = 0);
```

Comment:
I think that (just like for e.g. `std::string::substr`) it would be more uniform to use `size_t(-1)` as the default value meaning the whole stream.